app/src/main/java/com/io/moviO/movieDetails/MovieDetailsFragment.kt

```diff
@@ -23,15 +24,16 @@ class MovieDetailsFragment : Fragment(R.layout.fragment_movie_details) {
         requireArguments().getString(ARG_MOVIE_ID)?.let { viewModel.getMovieById(it) }
         viewModel.movie.observe(viewLifecycleOwner, Observer {
             when (it) {
-                is DataResult.Success ->
+                is DataResult.Success -> {
                     binding.apply {
                         movieName.text = it.value.name
-                        moviePoster.setBackgroundResource(it.value.poster)
                         movieYear.text = it.value.year
                         movieGenre.text = it.value.gerne
                         movieCast.text = it.value.cast
                         overviewText.text = it.value.overview
                     }
+                    Glide.with(this).load(it.value.poster).into(binding.moviePoster)
```

Comment:
this could be in the apply scope. The problem you faced is `this` within the scope is the `binding` so you need to rework it to `Glide.with(this@MovieDetailsFragment).load(it.value.poster).into(moviePoster)`

Reply:
resolved